### ISSUE TITLE
Errors on current version RPi Zero W installation

### ISSUE THREAD

**Issue by BradAndAU** [andrewjfreyer/monitor#175]
Followed the installation guide and get these errors, mac addresses removed;

> starting monitor.sh (v. 0.2.150)...
> removing web request caches
mkfifo: cannot create fifo 'packet_pipe': File exists
> 00:00:00:00:00:00 confidence topic: monitor/rpi0w/00:00:00:00:00:00 (has not previously connected to hci0)
> 00:00:00:00:00:00 confidence topic: monitor/rpi0w/00:00:00:00:00:00 (has not previously connected to hci0)
[+] 0.2.150 29-04-2019 05:13:32 PM [CMD-NAME]	00:00:00:00:00:00 White mi  Xiaomi Communications Co Ltd
*** Error in `/usr/bin/mosquitto_pub': munmap_chunk(): invalid pointer: 0xbed8c7cb ***
./support/mqtt: line 46:  2153 Aborted                 $mosquitto_pub_path -I "$mqtt_publisher_identity" $mqtt_version_append $mqtt_ca_file_append -L "$mqtt_url$mqtt_topicpath/$mqtt_publisher_identity/status" -m "online" 2>&1
[+] 0.2.150 29-04-2019 05:13:35 PM [CMD-INFO]	**** started arrival scan [x1 max rep] **** 
*** Error in `/usr/bin/mosquitto_pub': munmap_chunk(): invalid pointer: 0xbed0b7cb ***
./support/mqtt: line 46:  2685 Aborted                 $mosquitto_pub_path -I "$mqtt_publisher_identity" $mqtt_version_append $mqtt_ca_file_append -L "$mqtt_url$mqtt_topicpath/$mqtt_publisher_identity/status" -m "online" 2>&1
[+] 0.2.150 29-04-2019 05:13:42 PM [CMD-POST]	monitor/rpi0w/00:00:00:00:00:00
*** Error in `/usr/bin/mosquitto_pub': munmap_chunk(): invalid pointer: 0xbea536e0 ***
./support/mqtt: line 184:  2712 Aborted                 ( $mosquitto_pub_path -I "$mqtt_publisher_identity" $should_retain $mqtt_version_append $mqtt_ca_file_append -L "$mqtt_url$topic" -m "$message" 2>&1 )

Found a thread about this on HA forums and followed post by davst to git checkout 05500a8eb354a2db7302117124977490af5eaa4b which has resolved the issue.

Noticed you asked for this to be opened on github but couldn't see it so apologies if I've doubled up here.

Thanks Andrew.

**Comment by andrewjfreyer** [comment]
Thanks for opening an issue. What version of mosquitto are you running?

**Comment by BradAndAU** [comment]
Mosquitto version is 1.6.1

**Comment by andrewjfreyer** [comment]
Ok the only difference between the installation I just flashed to a spare pi and a spare pi3 (both of which are working as expected) is the mosquitto version. The image I'm using has mosquitto 1.5. I'm updating to 1.6 now to see if this is the cause.

**Comment by andrewjfreyer** [comment]
Confirmed that mosquitto 1.6 triggered the problem. My monitor nodes are all down after updating.

**Comment by andrewjfreyer** [comment]
Confirmed that the issue is on the eclipse end. Opened issue on eclipse/mosquitto [#1248.](https://github.com/eclipse/mosquitto/issues/1248). 

In the interim, the error can be ignored. All mqtt messages are still sent to the broker, so `monitor` will continue to function as expected.

**Comment by BradAndAU** [comment]
Thanks for the quick response Andrew, I’ll flash back to 0.2.150 and test.

**Comment by andrewjfreyer** [comment]
Best solution right now until this is fixed on mosquitto's end is to roll back to the previous version of 1.5.8:

```bash
sudo apt-get remove libmosquitto-dev mosquitto mosquitto-clients libmosquitto1
sudo apt-get purge libmosquitto-dev mosquitto mosquitto-clients libmosquitto1
sudo apt-get autoremove --purge

sudo apt-get install -f libmosquitto-dev=1.5.8-0mosquitto1 mosquitto=1.5.8-0mosquitto1 mosquitto-clients=1.5.8-0mosquitto1 libmosquitto1=1.5.8-0mosquitto1
```

**Comment by andrewjfreyer** [comment]
mosquitto has fixed this in a [release set up for tomorrow.](https://github.com/eclipse/mosquitto/issues/1248#issuecomment-487769676) very fast response!

**Comment by BradAndAU** [comment]
Amazing! Thanks Andrew

**Comment by andrewjfreyer** [comment]
Confirmed fixed as of mosquitto 1.6.2, released today!

```bash
sudo apt-get update
sudo apt-get upgrade
```
